From d2cb05093a5038deb80a09b29dc0175935857a31 Mon Sep 17 00:00:00 2001
From: Pengfei Xu <pengfei.xu@intel.com>
Date: Thu, 11 Jan 2024 11:06:36 +0100
Subject: [PATCH] keyctl05: Improve the dns_res_payload for boundary testing

kernel commit acc657692aed ("keys, dns: Fix size check of V1 server-list
header") from upcoming v6.8-rc1 (going to be backported to stable),
which fixes the add_key syscall incoming data boundary size issue, 6
bytes incoming data is now enough.

Link: https://lore.kernel.org/ltp/20240111100636.1897637-1-pvorel@suse.cz/

Upstream-Status: Backport from
[https://github.com/linux-test-project/ltp/commit/4e940005]

Reviewed-by: Petr Vorel <pvorel@suse.cz>
Signed-off-by: Pengfei Xu <pengfei.xu@intel.com>
[ pvorel: add acc657692aed, reword commit message ]
Signed-off-by: Petr Vorel <pvorel@suse.cz>

[xchen: Adapted the patch for context changes.]
Signed-off-by: Xiangyu Chen <xiangyu.chen@windriver.com>
---
 testcases/kernel/syscalls/keyctl/keyctl05.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/testcases/kernel/syscalls/keyctl/keyctl05.c b/testcases/kernel/syscalls/keyctl/keyctl05.c
index 7d7c076c0..b71f111fd 100644
--- a/testcases/kernel/syscalls/keyctl/keyctl05.c
+++ b/testcases/kernel/syscalls/keyctl/keyctl05.c
@@ -198,8 +198,12 @@ static void do_test(unsigned int i)
 	/*
 	 * We need to pass check in dns_resolver_preparse(),
 	 * give it dummy server list request.
+	 * From v6.8-rc1 commit acc657692aed438e9931438f8c923b2b107aebf9:
+	 * the incoming data for add_key() sysdall should be not less than 6
+	 * bytes, because struct dns_server_list_v1_header is 6 bytes.
+	 * The minimum payload will be tested here for boundary testing.
 	 */
-	static char dns_res_payload[] = { 0x00, 0x00, 0x01, 0xff, 0x00 };
+	static char dns_res_payload[] = { 0x00, 0x00, 0x01, 0xff, 0x00, 0x00 };
 
 	switch (i) {
 	case 0:
@@ -223,6 +227,7 @@ static struct tst_test test = {
 	.forks_child = 1,
 	.tags = (const struct tst_tag[]) {
 		{"linux-git", "63a0b0509e70"},
+		{"linux-git", "acc657692aed"},
 		{}
 	}
 };
-- 
2.34.1

